app/views/trains/new.html.slim

```diff
@@ -1,2 +1,3 @@
 h1 Create train
+= render partial: 'form', locals: { checked: train ? train["active"]=="true" : false, button: "Create", url: trains_path, method: :post }
```

Comment:
where is the controller action for this view.
also please dont do the checked thing, this code look very confusing and could have been handled with defaults on an instantiated object instead of this boolean inside a boolean checking ternary which is very very confusing.

Reply:
looking into this, i didnt know that if you dont define a controller action but have views and routes for the action it will just try to render the defaults.

But since this isnt instantly aparant and would be easy to miss (like right now haha) we should probably define the new even if its empty.

but i would still recommend having the new setup the initial train object so we can avoid all the gross ternaries.

so having something like:

```
def index
  @train = { "active": false }
end
```

Reply:
> but i would still recommend having the new setup the initial train object so we can avoid all the gross ternaries.

the reason for the strange `checked` value was because when new train, the default `active` will be false which is correct as the default `active` value

then, when i submit the form, the errors show, however, the `active` radio button isnt checked on false anymore which shouldnt be the case

Reply:
so instead i used `checked: train['active']=="true"` in the `new` form locals which i think fixes the problem

Reply:
boolean values shouldnt be submitting as a string.
`train['active']` should be `true` not `'true'` they are different.
so doing `train['active']=="true"` is wrong as `true` would become `false`

booleans man, they are going to become your friend.

your params should also be submitting the value as a boolean too, you can check this in your controller pretty easily too

Reply:
> boolean values shouldnt be submitting as a string.
> `train['active']` should be `true` not `'true'` they are different.
> so doing `train['active']=="true"` is wrong as `true` would become `false`
> 
> booleans man, they are going to become your friend.
> 
> your params should also be submitting the value as a boolean too, you can check this in your controller pretty easily too

so the reason why i used `'true'` instead of `true` in the new locals, 
is because when i used `true`, 
i clicked the new train and the default active is false which is correct,

however, when i changed active to true and click submit, 
the errors show and the new form renders
but the active status reverts back to false, which should not be the case

Reply:
if the status goes back to the default, then the updated `@train` when create fails is the issue. so fix that imo

Reply:
https://github.com/mmmmaximus/rails-truck-logistics/pull/36/files#diff-a7ee1eaf7e61cb55a8c25d72fbf7a671bf38c47521d89874b373295771f362daR21

Reply:
a string for a boolean is just wrong, we dont want params for boolean values to be strings.
if you do `active["true"]` and the value is `true` which is boolean, and not `'true'` which is a string.
when you do `true == 'true'` it would return false. which is wrong.
when dealing with booleans we should let them tell us what they are.
so instead of doing comparisons on `active["true"] === 'true'` we can just do `active["true"]` as the value is a boolean so we dont need comparison

Reply:
> if you do `active["true"]` and the value is `true` which is boolean, and not `'true'` which is a string.

```
 def create
    response = TrainApi.new.create(train_params)

    if response.success?
      redirect_to trains_path, notice: response["message"]
    else
      flash["errors"] = response["errors"]
      @train = params[:train]
      debugger
      render "new"
    end
  end
```
in my create method, when it hits the debugger with an empty form,
`params[:train]["active"]` turns out to be a string instead of a boolean

could i use a simple if else to convert the string to boolean in the create method?

Reply:
https://stackoverflow.com/questions/19339137/how-to-set-boolean-values-to-radio-buttons-in-rails-4-form
how about something like this if its submitting as a string

Reply:
> https://stackoverflow.com/questions/19339137/how-to-set-boolean-values-to-radio-buttons-in-rails-4-form
> how about something like this if its submitting as a string

just tried some of the solutions,
it keeps saving `train["active"]` as a string regardless